lib/msf/core/db_manager/service.rb

```diff
@@ -131,15 +135,31 @@ def report_service(opts)
   end
 
   # Returns a list of all services in the database
-  def services(wspace = workspace, only_up = false, proto = nil, addresses = nil, ports = nil, names = nil)
+  def services(opts)
+    opts.delete(:workspace) # Mdm::Service apparently doesn't have an upstream Mdm::Workspace association
+
+  ::ActiveRecord::Base.connection_pool.with_connection {
+    search_term = opts.delete(:search_term)
+    only_up = opts.delete(:only_up)
+    opts[:state] = [Msf::ServiceState::Open] if only_up
+    opts["hosts.address"] = opts.delete(:addresses)
```

Comment:
Would it be possible to leverage `Mdm::Host.arel_table[:address]` for the condition?